### PR TITLE
Implement iexec.account.approve()

### PR DESCRIPTION
**Pull request by abbesBenayache**

### CODE REVIEW

File: test/lib/e2e/IExecAccountModule.test.js [hunk 1]
```diff
@@ -352,4 +353,54 @@ describe('account', () => {
       );
     });
   });
+
+  describe('approve()', () => {
+    test('require a signer', async () => {
+      const spenderAddress = Wallet.createRandom().address;
```
Comment (PierreJeanjacquot) [inline test/lib/e2e/IExecAccountModule.test.js:359]: use `getRandomAddress()` from utils (factorized version of this code)

File: test/lib/e2e/IExecAccountModule.test.js [hunk 2]
```diff
@@ -352,4 +353,54 @@ describe('account', () => {
...
+    });
+
+    test('prevents approve 0', async () => {
+      const { iexec, wallet } = getTestConfig(iexecTestChain)();
+      const spenderAddress = wallet.address;
+
+      await expect(iexec.account.approve(0, spenderAddress)).rejects.toThrow(
+        Error('Approve amount must be greater than 0'),
+      );
+    });
```
Comment (PierreJeanjacquot) [inline test/lib/e2e/IExecAccountModule.test.js:397]: replace by prevent approve negative amount

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]

File: src/common/account/allowance.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        strict: true,
+      });
+    }
+
+    const vSpenderAddress = await addressSchema({
+      ethProvider: contracts.provider,
+    }).validate(spenderAddress);
+
+    const iexecContract = contracts.getIExecContract();
+    const tx = await iexecContract.approve(vSpenderAddress, vAmount);
```
[the remark this answers is not in the document]
Comment (abbesBenayache) [reply src/common/account/allowance.js:35]: commit : 7b8b3cd71e01f80e0fb5eacf599b76e166e7df90

File: docs/classes/IExecAccountModule.md
[the remark this answers is not in the document]
Comment (abbesBenayache) [reply docs/classes/IExecAccountModule.md:1]: commit : cfac1c60a91f5f8d7a7ed94b7762a5f0a842f3ad

File: src/common/account/allowance.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  try {
+    checkSigner(contracts);
+    const vAmount = await nRlcAmountSchema().validate(amount);
+    if (new BN(vAmount).lte(new BN(0)))
+      throw Error('Approve amount must be greater than 0');
+    if (contracts.flavour === 'enterprise') {
+      await isInWhitelist(contracts, await getAddress(contracts), {
+        strict: true,
+      });
+    }
```
[the remark this answers is not in the document]
Comment (abbesBenayache) [reply src/common/account/allowance.js]: commit : 0395df235f3f4a19db8c0e963385d87dffebb247

File: test/lib/e2e/IExecAccountModule.test.js (hunk 2 above)
Comment (abbesBenayache) [reply test/lib/e2e/IExecAccountModule.test.js]: commit : 00421ba2ab1b46418605f948913199bdc31432e0

**Review by PierreJeanjacquot — APPROVED** [review]

**Pull request merged**